### PR TITLE
Add Docker OSS/Enterprise quickstarts

### PR DESCRIPTION
**Pull request by webvictim**
Add some information to both OSS and Enterprise quickstarts on Docker image naming/base images. Provide a "quickstart" docker-compose YAML file for each to get people going quickly.

### CODE REVIEW

File: docs/4.3/enterprise/quickstart-enterprise.md [hunk 1]
```diff
@@ -315,6 +320,101 @@ Any SAML-compliant provider can be configured with Teleport by following the
 same steps.  There are Teleport Enterprise customers who are using Oracle IDM,
 SailPoint and others.
 
+## Run Teleport Enterprise using Docker
+
+We provide pre-built Docker images for every version of Teleport Enterprise. These images are hosted on quay.io.
+
+- [All tags under `quay.io/gravitational/teleport-ent` are Teleport Enterprise images](https://quay.io/repository/gravitational/teleport-ent?tag=latest&tab=tags)
```
Comment (awly) [inline docs/4.3/enterprise/quickstart-enterprise.md:327]: nit: should the link be this whole sentence or just the image name?

File: docs/4.3/enterprise/quickstart-enterprise.md [hunk 2]
```diff
@@ -315,6 +320,101 @@ Any SAML-compliant provider can be configured with Teleport by following the
...
+
+We currently only offer Docker images for `x86_64` architectures.
+
+### Pick your image
+
+This table gives an idea of how our image naming scheme works. We offer images which point to a static version of Teleport Enterprise, as well as images which are
+automatically rebuilt every night. These nightly images point to the latest version of Teleport Enterprise from the current release branch, plus the two previous release branches.
```
Comment (awly) [inline docs/4.3/enterprise/quickstart-enterprise.md:334]: nit: how about "from the three most recent release branches"?

File: docs/4.3/enterprise/quickstart-enterprise.md [hunk 3]
```diff
@@ -315,6 +320,101 @@ Any SAML-compliant provider can be configured with Teleport by following the
 same steps.  There are Teleport Enterprise customers who are using Oracle IDM,
 SailPoint and others.
 
+## Run Teleport Enterprise using Docker
```
Comment (awly) [inline docs/4.3/enterprise/quickstart-enterprise.md:323]: Add a note (somewhere in this section) that this won't let you SSH onto the host. It's probably not intended for prod usage on nodes.

File: docker/teleport-ent-quickstart.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    command: -c "if [ ! -f /etc/teleport/teleport.yaml ]; then /usr/local/bin/teleport configure > /etc/teleport/teleport.yaml; fi"
+    volumes:
+      - ./teleport/config:/etc/teleport
+
+  teleport:
+    image: quay.io/gravitational/teleport-ent:4.3
+    container_name: teleport
+    entrypoint: /bin/sh
+    hostname: localhost
+    command: -c "sleep 1 && /usr/local/bin/teleport start"
```
[the remark this answers is not in the document]
Comment (webvictim) [reply docker/teleport-ent-quickstart.yml:17]: Yeah - the `teleport-configure` container just writes the config file and exits. In lieu of a proper sequential start in `docker-compose`, this is just to make sure that if they start at the same time then the config file gets written before the other container needs it.

File: docker/teleport-ent-quickstart.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    ports:
+      - "3023:3023"
+      - "3025:3025"
+      - "3080:3080"
```
[the remark this answers is not in the document]
Comment (webvictim) [reply docker/teleport-ent-quickstart.yml:21]: I guess this wasn't intended to be a full reference production setup - more of a simple "here's how you can run a Teleport instance quickly on your local machine and test it out". I wanted to keep the same ports between `docker-compose` and `docker run` and having 6 ports on the command line seemed a bit over the top. I can add them if we think it's better to handle all possibilities.

File: docs/4.3/enterprise/quickstart-enterprise.md (hunk 2 above)
Comment (webvictim) [reply docs/4.3/enterprise/quickstart-enterprise.md:334]: Much better.

File: docs/4.3/enterprise/quickstart-enterprise.md (hunk 1 above)
Comment (webvictim) [reply docs/4.3/enterprise/quickstart-enterprise.md:327]: I think I was going for making it very obvious that it was a clickable link, as code blocks have different formatting.

File: docs/4.3/quickstart.md [hunk 6]
```diff
@@ -274,6 +279,91 @@ $ echo "Awesome!"
...
+| Image name | Community or Enterprise? | Teleport version | Image automatically updated? | Image base |
+|---|---|---|---|---|
+| `quay.io/gravitational/teleport:4.3` | Community | The latest version of Teleport Community 4.3 | Yes | [Ubuntu 20.04](https://hub.docker.com/_/ubuntu) |
+| `quay.io/gravitational/teleport:4.3.0` | Community | 4.3.0 | No | [Ubuntu 18.04](https://hub.docker.com/_/ubuntu) |
+
+For testing, we always recommend that you use the latest release version of Teleport, which is currently `{{teleport.latest_oss_docker_image}}`.
+
+### Quickstart using docker-compose
+
+The easiest way to start Teleport quickly is to use `docker-compose` with our [`teleport-quickstart.yml`](https://github.com/gravitational/teleport/blob/master/docker/teleport-quickstart.yml) file:
```
Comment (benarent) [inline docs/4.3/quickstart.md:305]: To be extra explicit, we should note that the user needs to install Docker on their machine.

File: docs/4.3/enterprise/quickstart-enterprise.md [hunk 7]
```diff
@@ -41,6 +41,11 @@ _"node.example.com"_  | 10.1.1.11      | This server will only run the SSH servi
 This Quick Start Guide assumes that both servers are running a [systemd-based](https://www.freedesktop.org/wiki/Software/systemd/)
 Linux distribution such as Debian, Ubuntu or a RHEL derivative.
 
+## Optional: Quickstart using Docker
```
Comment (benarent) [inline docs/4.3/enterprise/quickstart-enterprise.md:44]: Longer term we'll split this out, but I think it's ok to add into the quickstart for now.

File: docs/4.3/enterprise/quickstart-enterprise.md [hunk 8]
```diff
@@ -315,6 +320,104 @@ Any SAML-compliant provider can be configured with Teleport by following the
...
+
+```bash
+docker exec teleport tctl users add testuser --roles=admin
+```
+
+When you run this command, Teleport will output a URL which you must open to complete the user signup process:
+
+```bash
+User testuser has been created but requires a password. Share this URL with the user to complete user setup, link is valid for 1h0m0s:
+https://localhost:3080/web/invite/4f2718a52ce107568b191f222ba069f7
```
Comment (benarent) [inline docs/4.3/enterprise/quickstart-enterprise.md:413]: Does the user require any extra steps to setup TLS for Docker on Local host?

**Review by benarent — CHANGES_REQUESTED** [review]

File: docs/4.3/enterprise/quickstart-enterprise.md (hunk 8 above)
Comment (awly) [reply docs/4.3/enterprise/quickstart-enterprise.md:413]: I don't think so.
Docker container just exposes a TCP port. What runs on top of that should be invisible to docker.

File: docker/teleport-ent-quickstart.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (awly) [reply docker/teleport-ent-quickstart.yml:17]: Leave a comment (either on `teleport-configure` or here) for a future reader ;)

File: docker/teleport-ent-quickstart.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (awly) [reply docker/teleport-ent-quickstart.yml:21]: Sounds reasonable, let's keep this simple

File: docs/4.3/quickstart.md [hunk 9]
```diff
@@ -274,6 +279,91 @@ $ echo "Awesome!"
...
+We provide pre-built Docker images for every version of Teleport. These images are hosted on quay.io.
+
+- [All tags under `quay.io/gravitational/teleport` are Teleport Community images](https://quay.io/repository/gravitational/teleport?tag=latest&tab=tags)
+
+We currently only offer Docker images for `x86_64` architectures.
+
+### Pick your image
+
+This table gives an idea of how our image naming scheme works. We offer images which point to a static version of Teleport, as well as images which are
+automatically rebuilt every night. These nightly images point to the latest version of Teleport from the current release branch, plus the two previous release branches.
```
Comment (awly) [inline docs/4.3/quickstart.md:293]: Copy the sentence from enterprise version.
Maybe the whole thing should link here? And the enterprise version only explain the image selection + license file?

Comment (webvictim) [reply docs/4.3/quickstart.md:293]: Unfortunately the `tctl users add` syntax is different for OSS versus Enterprise, plus I had to do some extra fudging to get the license file to work for Enterprise. Painful as it is, it's probably best to keep them separate.

File: docs/4.3/enterprise/quickstart-enterprise.md (hunk 8 above)
Comment (webvictim) [reply docs/4.3/enterprise/quickstart-enterprise.md:413]: Yeah, it's likely that the user will get an insecure warning in their browser that they'll have to click through but that's the case with regular Teleport setups too.

Comment (benarent) [reply docs/4.3/enterprise/quickstart-enterprise.md:419]: How about we recommend / add a note about using https://github.com/FiloSottile/mkcert ?

**Comment by webvictim** [comment]
Please review again @benarent when you have a chance. Think I addressed your comments.

**Review by awly — APPROVED** [review]

**Review by benarent — APPROVED** [review]
I'll approve and test this, and loop back about the localhost cert issue. Which is an issue for our normal quickstart so not really a blocker.

**Comment by webvictim** [comment]
@benarent Do you want me to wait for your test before merging?

**Comment by benarent** [comment]
No waiting needed, I'll merge and deploy this now.

**Comment by webvictim** [comment]
Thanks!

**Pull request merged**